### PR TITLE
Setup graceful handler for SIGINT/SIGTERM [SITL-62]

### PR DESCRIPTION
**Pull request by silverjam**
Use warp "graceful shutdown" features so that the SIGTERM received from kubernetes will initiate a clean shutdown of connections.  We'll still need to configure timeouts in kubernetes to something longer than the default (60 seconds), the graceful shutdown will not allow new connections, but will allow existing downloads/browsing to continue until they complete.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -158,12 +158,11 @@ async fn main() -> Result<()> {
     let credentials_provider = DefaultCredentialsProvider::new().unwrap();
     let s3 = S3Client::new_with(http_client, credentials_provider, Region::default());
 
-    setup_cancel_handler();
-
     use Command::*;
 
     match cli.cmd {
         Put { bucket, key, file } => {
+            setup_upload_ctrlc_handler();
```
Comment (betog25) [inline src/main.rs:165]: I'm just curious what the `ctrlc` stands for/means.  Controller?

Comment (betog25) [reply src/main.rs:165]: Nevermind, I found this https://docs.rs/ctrlc/3.1.7/ctrlc/

**Review by betog25 — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (silverjam) [reply src/main.rs]: fixed

**Comment by silverjam** [comment]
@notoriaga merging, if you have any comments I can fix in another PR

**Pull request merged**